### PR TITLE
[DO NOT MERGE] Add local jQuery

### PR DESCRIPTION
**Pull request by andysellick**
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

## What
Adds jQuery into this application, using the package.json to include the version currently in the components gem.

**This PR should be merged immediately prior to the removal of jQuery from the rest of GOV.UK, as the change will result in an increased asset size for this application**. Having two copies of jQuery running together appears to not cause any problems, but we want to avoid a situation where collections-publisher has no jQuery supplied.

Also fixes a problem where the autocomplete component wasn't appearing correctly in the component guide for this application.

## Why
We're removing jQuery from the components gem. This application has a lot of jQuery and isn't public facing, so to prevent  blocking it's easiest right now to give collections-publisher its own jQuery.

## Visual changes
None

Trello card: https://trello.com/c/mAfQNVRG/745-investigate-which-apps-we-could-remove-jquery-from?filter=member:andysellick2

### CODE REVIEW

**Review by leenagupte — COMMENTED** [review]
It looks good to me, though I'm not sure how to test that the auto-complete still works. Were you able to test this locally.

**Comment by andysellick** [comment]
> It looks good to me, though I'm not sure how to test that the auto-complete still works. Were you able to test this locally?

I was. Easy way to do it is to copy the example code from the component guide into any template that matches a page that can be rendered locally.

File: app/assets/javascripts/components/autocomplete.js [hunk 1]
```diff
@@ -3,11 +3,11 @@ window.GOVUK = window.GOVUK || {}
...
 (function (Modules) {
-  function Autocomplete () { }
-
-  Autocomplete.prototype.start = function ($module) {
-    this.$module = $module[0]
+  function Autocomplete ($module) {
+    this.$module = $module
+  }
 
+  Autocomplete.prototype.init = function () {
```
[the remark this answers is not in the document]
Comment (andysellick) [reply app/assets/javascripts/components/autocomplete.js:10]: curated_lists (and the other similar JS in this repo) has an `init` function, but it isn't one of our modules, so it isn't initialised by the central modules startup code in the components gem and doesn't require any modification. Here's curated lists:

```
(function () {
  'use strict'
  window.GOVUK = window.GOVUK || {}
  var $ = window.jQuery
  var csrfToken = $('meta[name="csrf-token"]').attr('content')

  GOVUK.curatedLists = {
    init: function () {
      // etc.
```

and here's a typical module:

```
window.GOVUK = window.GOVUK || {}
window.GOVUK.Modules = window.GOVUK.Modules || {};

(function (Modules) {
  function Autocomplete ($module) {
    this.$module = $module
  }

  Autocomplete.prototype.init = function () {
   // etc.
```

Note the lack of constructor in the first example (`function NameOfModule($module) {}`) and the lack of any mention of `window.GOVUK.Modules`.

[the remark this answers is not in the document]
Comment (leenagupte) [reply app/assets/javascripts/components/autocomplete.js:10]: Thank you!

**Comment by kevindew** [comment]
I don't think you need to bundle a version of jQuery for this app.

It's a bit of a confusing app because it's half govuk_admin_template, half govuk_publishing_components. The govuk_admin_template side should automatically include the jQuery from govuk_admin_template, whereas - hopefully - the govuk_publishing_components side doesn't need jQuery. 

I'm just giving this a spin locally to work out what's going on because there seems to be imports in the different files that I think shouldn't be there, but I could be wrong and there is some lingering dependencies scattered between the two.

**Comment by leenagupte** [comment]
> > It looks good to me, though I'm not sure how to test that the auto-complete still works. Were you able to test this locally?
> 
> I was. Easy way to do it is to copy the example code from the component guide into any template that matches a page that can be rendered locally.

That's a great idea. It works like a charm! 🎉

**Comment by andysellick** [comment]
> I don't think you need to bundle a version of jQuery for this app.

@kevindew just checked this locally with the branch of the [components gem without jQuery](https://github.com/alphagov/govuk_publishing_components/pull/2613) and got jQuery related errors on the first page. Checking the page source doesn't show any inclusion of a local jQuery (the errors are caused by jQuery-UI trying to initialise without jQuery being included).

Screenshot 2022-03-23 at 11 23 45

**Comment by DilwoarH** [comment]
> > I don't think you need to bundle a version of jQuery for this app.
> 
> @kevindew just checked this locally with the branch of the [components gem without jQuery](https://github.com/alphagov/govuk_publishing_components/pull/2613) and got jQuery related errors on the first page. Checking the page source doesn't show any inclusion of a local jQuery (the errors are caused by jQuery-UI trying to initialise without jQuery being included).
> 
> Screenshot 2022-03-23 at 11 23 45

I think this is because of this: https://github.com/alphagov/collections-publisher/blob/main/app/assets/javascripts/admin_layout.js#L4

I believe we initially needed this for the reorderable component which has since moved to the publishing components gem and doesn't need jQuery UI anymore.

File: app/assets/javascripts/admin_layout.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
+//= require jquery/dist/jquery
```
Comment (DilwoarH) [inline app/assets/javascripts/admin_layout.js:1]: I don't think we need jQuery (as far as I am aware)

File: app/assets/javascripts/admin_layout.js [hunk 3]
```diff
@@ -1,5 +1,7 @@
...
 //= require govuk_publishing_components/dependencies
 //= require govuk_publishing_components/all_components
+//= require components/autocomplete
 
 //= require jquery-ui.sortable.min
 //= require jquery_ujs
```
Comment (DilwoarH) [inline app/assets/javascripts/admin_layout.js:7]:
```suggestion

```

Comment (andysellick) [reply app/assets/javascripts/admin_layout.js:7]: Hmm just tried that and I'm still getting errors from missing jQuery.

**Comment by andysellick** [comment]
I'm going to close this PR in favour of https://github.com/alphagov/collections-publisher/pull/1550, thanks all for the reviews and comments.

**Pull request closed without merging**